EventHandlers/JoinOptinGuildHandler.cs

```diff
@@ -30,12 +30,15 @@ public class JoinOptinGuildHandler : AbstractRegexHandler
 
         /// <summary>
         /// The matcher for detecting the phrases:
-        /// - <@123> join {1}
-        /// - <@123> join #{1}
-        /// And captures the proposed channel name into group 1
+        /// - <@123> join {channel}
+        /// - <@123> join #{channel}
+        /// - <@123> join opt-in {channel}
+        /// - <@123> join opt in {channel}
+        /// - <@123> join optin {channel}
+        /// And captures the proposed channel name into the named capture group "channel"
         /// </summary>
         private static Regex matcher = new Regex(
-            @"^\s*\<\@\!?\d+\>\s*join\s+#?(\S+)$",
+            @"^\s*\<\@\!?\d+\>\s*join(\s+opt(?:-|\s)?in)?\s+#?(?<channel>\S+)$",
```

Comment:
I had no idea you could name groups, this is cool! We'll have to figure out a solution to the name spacing problem when I implement invite only channels later though.

Reply:
Indeed! I imagine you won't be able to join invite only channels via bot commands, so that shouldn't be an issue. It just means the leave command will never able to be just "leave <channel>" unless we want to try to handle the case when an opt-in and invite only channel have the same name.